src/main/java/org/example/api/SystemApi.java

```diff
@@ -0,0 +1,36 @@
+package org.example.api;
+import org.example.model.Directory;
+import org.example.model.MyFile;
+
+public class SystemApi {
+    private final IFileApi api = new FileService();
+
+    public MyFile addFile(Directory directory, MyFile myFile) {
+        return api.addFile(directory, myFile);
+    }
+
+    public MyFile removeFile(Directory directory, String fileName) {
```

Comment:
we can rewrite parameter and response types of all these functions to be simpler. instead of directory it can be just a String path, instead of MyFile response it can be a new MyResponse object which will contain fileId(I'll explain this id concept later, in the comments), fileContent this could be byte[] or just String. and String enum errorCode. This error codes are needed to udentify exception situations, what if file was not found? what if some random exception happened in the code? what if file already exists with same name in the folder? we want user who calls this API know exactly what happened?  Therefore based on service response we construct Api response object which goes to the caller.

Reply:
I made some modifications to keep the logic, I didn't change the return types, because if I return Strings or byte arrays from my methods in that case readFile method becomes useless, but we obviously need that, so for users there is one more step to get the content by using readFile method